checkmarx/model.py

```diff
@@ -114,7 +116,33 @@ class ScanResult:
     """
     ScanResult is a data container for a scan result for a component version
     """
+    artifact_name: str
     project_id: int
-    component: cm.Component
     scan_response: ScanResponse
     scan_statistic: ScanStatistic
+
+
+@dataclasses.dataclass
+class FinishedScans:
+    failed_scans: typing.List[str] = dataclasses.field(default_factory=list)
+    scans_above_threshold: typing.List[ScanResult] = dataclasses.field(default_factory=list)
+    scans_below_threshold: typing.List[ScanResult] = dataclasses.field(default_factory=list)
+
+
+# abstraction of component model v2 source and resource
+@dataclasses.dataclass
+class ScanArtifact:
+    name: str
+    access: typing.Union[
+        cm.OciAccess,
+        cm.GithubAccess,
+        cm.HttpAccess,
+        cm.ResourceAccess,
+        None
```

Comment:
is it really a sane thing to have an artifact w/o an access(or)?

Reply:
note to self: we should have called them `accessor`s :-/

Reply:
no I think it's not good to have that. I will remove None